src/api/query.rs

```diff
@@ -82,13 +83,22 @@ pub trait SimpleQueryHandler: Send + Sync {
             }
         }
 
-        client
-            .feed(PgWireBackendMessage::ReadyForQuery(ReadyForQuery::new(
-                TransactionStatus::Idle,
-            )))
-            .await?;
-        client.flush().await?;
-        client.set_state(super::PgWireConnectionState::ReadyForQuery);
+        match client.state() {
+            PgWireConnectionState::CopyInProgress => {
```

Comment:
Checked again, I think we can change the state to `CopyInProgress` if any of `Response::CopyIn`, `Response::CopyOut` or `Response::CopyBoth` is returned.

Reply:
Done.